README.md

```diff
@@ -22,12 +27,370 @@ Download the collection and extract it in the `data` directory, under a `collect
+
+### Inversed index construction
+
+Building the inversed index consist on applying these differents steps to the whole collection, and then iterate over all documents and tokens, counting and storing the index in a dictionnary.
+
+We support three index types:
+
+- *document index*: For each term of the collection, it returns the ids of the documents in which the term appears.
+
+```python
+{"information": [1, 4]} # the term 'information' appears in docs 1 and 4
+```
+
+- *frequency index*: For each term of the collection, for each document in which the term appears, it returns both the id of the document and the term frequency of the term in the document.
+
+```python
+{"information": {1: 2, 4: 19}} # the term 'information' appears in docs 1 (2 times) and 4 (19 times)
+```
+
+- *position index*, For each term of the collection, for each document in which the term appears, it returns both the id of the document and the positions of the term in the document.
+
+```python
+{"information": {1: [1, 19], 4: [0, 2, 5]}} # the term 'information' appears in docs 1 (at position 1 and 19) and 4 (at position at 0, 2 and 5)
+```
+
+To create an index, a cli is provided with `[src/preprocess.py](src/preprocess.py)` :
+
+```bash
+> python src/preprocess.py --help
+usage: preprocess.py [-h] [--pos POS] index_type output
+
+positional arguments:
+  index_type  type of the index to build
+  output      path where the index will be saved
+
+optional arguments:
+  -h, --help  show this help message and exit
+  --pos POS   use the the Part-Of-Speech (pos) lemmatization, or simple
+              stemmer (default=True)
+```
+
+For example to create an frequency index with the simple Snowball stemmer, saving it as a `pickle` binary, you can run `python src/preprocess.py 2 --pos False data/frequency_index_stem.pkl`.
+
+To reduce development time, we also have an environment variable for a `DEV` flag, which load less files from the corpus : `DEV=1 DEV_ITER=100 python src/preprocess.py 1`. By default, the index is saved in the `data` folder as `index.pkl` (or `dev_index.pkl`).
+
+### First analysis
+
+**Note: We're doing the time benchmark on our working machines. Therefore the resuts can vary quite a lot depending of the work actually done on the computer, process launched by the OS, etc... These benchmarks are only here to give an general idea of the steps which can take some time.**
+
+Let's benchmark a first run for our index creation, with an index of type 1, a complete lemmatization.
+
+| *Steps*    | Corpus loading | Tokenization | Stop words | Lemmatization | Stop words | Index  | Total   |
+|----------|----------------|--------------|------------|---------------|------------|--------|---------|
+| *Time (s)* | 12.38          | 80           | 67         | 679           | 49         | 2069   | **2956.38** |
+| *%*        | 0.42%          | 2.71%        | 2.27%      | 22.97%        | 1.66%      | 69.98% | **100.00%** |
```

Comment:
Are you sure you mesured the index construction on its own ? From what I saw the decorator was on the whole index function, which includes lemmatization. If it is not on its own, then the 23% of the lemmatiation are a part of the 70% of the index ...

Reply:
Yep i did this msitake at first, but then i used the tqdm part. This is an example with the very bad index type1 with a list.